### ISSUE TITLE
Github CI Gradle interface method overriding bug

### ISSUE THREAD

**Issue by Nephelite** [nus-cs2103-AY2223S1/forum#346]
Initially, my method's signature was
```void updateFilteredTaskList(Predicate<Task> predicate)```
in both the Model interface and the ModelManager class (public in the class).
This compiles fine in IntelliJ, so I pushed it and made a PR.
Github CI then tells me that it cannot compile because of the following error in the build:
![image](https://user-images.githubusercontent.com/97427604/196030928-417990ae-7175-4f52-8ae3-4372ea5baf66.png)
I can guarantee that the method is properly overridden. I can also 100% guarantee that there is no duplicate method. There is no other method with the same name.
I did not know what to do, so I tried a lot of things, and eventually, I just renamed the method to 
```void changeFilteredTaskList(Predicate<Task> predicate)```
and it worked. Github CI passed the check.

As a final test, I changed the method name back to update, and it fails again.

I suspect the reason is that I have another method in the class that is similar, it is updateFilteredPersonList. It has the same arguments too.

**Comment by damithc** [comment]
@Nephelite it might be useful to share the link to the PR so that others can have a look at the full code.

**Comment by Nephelite** [comment]
@damithc Sure, here are links to the relevant PRs
'Problematic' code:
https://github.com/AY2223S1-CS2103T-T13-4/tp/pull/152
Working code:
https://github.com/AY2223S1-CS2103T-T13-4/tp/pull/150

**Comment by damithc** [comment]
Let's wait to see if someone in class can figure out what's going on here ...

**Comment by samuelfangjw** [comment]
Hi! I'm a senior from a previous batch who chanced upon this forum post.

Let's dig a little deeper into this PR https://github.com/AY2223S1-CS2103T-T13-4/tp/pull/152.

We can view the parent of the head of the problematic branch. https://github.com/Nephelite/tp/blob/ac9539275389f6e0188debc519075dc96857dead/src/main/java/taskbook/model/Model.java. As expected, we don't find any sign of updateFilteredTaskList.

We can do the same for the head of the master branch we are trying to merge into. https://github.com/AY2223S1-CS2103T-T13-4/tp/blob/master/src/main/java/taskbook/model/Model.java. Seems like updateFilteredTaskList is already defined on line 152.

When we perform the merge, we end up with https://github.com/Nephelite/tp/blob/91139c011308b388b57168c8f8e7d3b141a8c860/src/main/java/taskbook/model/Model.java. Notice updateFilteredTaskList is defined on lines 128 and 152.

The final question is why GitHub did not flag this up when viewing changed files. The reason is it's doing a diff between the head of the problematic branch and the merge-base, not the latest commit on master.

TLDR, don't forget to keep your fork up to date with the upstream repo! Also consider applying the "Require branches to be up to date before merging" option in GitHub settings.

**Comment by Nephelite** [comment]
Thanks!
